homeassistant/components/homekit/accessories.py

```diff
@@ -93,9 +117,24 @@ def update_state_callback(self, entity_id=None, old_state=None,
     def update_state(self, new_state):
         """Method called on state change to update HomeKit value.
 
-        Overridden by accessory types.
+        Update battery service if available.
```

Comment:
Instead of using the `update_state` method, why not add a new one `update_battery_level` and call it before `update_state` in `update_state_callback`? That way you don't need to add `super` calls to every accessory type.